scripts/Interface/ui/drill/model/DrillExportModel.py

```diff
@@ -0,0 +1,103 @@
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+
+from mantid.simpleapi import mtd
+from mantid.kernel import config
+from mantid.api import WorkspaceGroup
+
+from .configurations import RundexSettings
+from .DrillAlgorithmPool import DrillAlgorithmPool
+from .DrillTask import DrillTask
+
+
+class DrillExportModel:
+
+    """
+    Dictionnary containing algorithms and activation state.
+    """
+    _exportAlgorithms = None
+
+    """
+    ThreadPool to run export algorithms asynchronously.
+    """
+    _pool = None
+
+    def __init__(self, acquisitionMode):
+        """
+        Create the export model by providing an aquisition mode.
+
+        Args:
+            acquisitionMode (str): acquisition mode
+        """
+        self._exportAlgorithms = {k:v
+                for k,v
+                in RundexSettings.EXPORT_ALGORITHMS[acquisitionMode].items()}
+        self._pool = DrillAlgorithmPool()
+
+    def getAlgorithms(self):
+        """
+        Get the list of export algorithm
+
+        Returns:
+            list(str): names of algorithms
+        """
+        return [algo for algo in self._exportAlgorithms.keys()]
+
+    def isAlgorithmActivated(self, algorithm):
+        """
+        Get the state of a specific algorithm.
+
+        Args:
+            algorithm: name of the algo
+        """
+        if algorithm in self._exportAlgorithms:
+            return self._exportAlgorithms[algorithm]
+        else:
+            return False
+
+    def activateAlgorithm(self, algorithm):
+        """
+        Activate a spefific algorithm.
+
+        Args:
+            algorithm (str): name of the algo
+        """
+        if algorithm in self._exportAlgorithms:
+            self._exportAlgorithms[algorithm] = True
+
+    def inactivateAlgorithm(self, algorithm):
+        """
+        Inactivate a specific algorithm.
+
+        Args:
+            algorithm (str): name of the algo
+        """
+        if algorithm in self._exportAlgorithms:
+            self._exportAlgorithms[algorithm] = False
+
+    def run(self, workspaceName):
+        """
+        Run the export algorithms on a workspace. If the provided workspace is
+        a groupworkspace, the export algorithms will be run on each member of
+        the group.
+
+        Args:
+            workspaceName (str): name of the workspace
+        """
+        exportPath = config.getString("defaultsave.directory")
+        tasks = list()
+        for wsName in mtd.getObjectNames():
+            if ((workspaceName in wsName)
+                    and (not isinstance(mtd[wsName], WorkspaceGroup))):
+                for a,s in self._exportAlgorithms.items():
+                    if s:
+                        filename = exportPath + wsName \
+                                   + RundexSettings.EXPORT_ALGO_EXTENSION[a]
+                        task = DrillTask(-1, a, InputWorkspace=wsName,
```

Comment:
Does the ref need to be something other than -1 here so that multiple export tasks can be distinguished from each other later on?

Reply:
The DrillTask ref is used to find out from which task success or error signals are received. This is used during the reduction to report the status on the table. But as I said in the comment, here we are ignoring these signals.